### PR TITLE
[CI] Fix nightly horovod test

### LINKED ISSUE #26440: [ml release] horovod train test is failing
### What happened + What you expected to happen

  File "train_horovod_multi_node_test.py", line 26, in <module>
    assert len(worker_result) == num_epochs
TypeError: object of type 'NoneType' has no len()

https://buildkite.com/ray-project/release-tests-branch/builds/764#0181e4fe-99ff-45df-840a-56e289a62e11

### Versions / Dependencies

Master

### Reproduction script

NA

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

### PR DESCRIPTION
**Pull request by Yard1**
## Related issue number

Closes https://github.com/ray-project/ray/issues/26440

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by xwjiang2010** [comment]
release test link: https://buildkite.com/ray-project/release-tests-pr/builds/8893#0181eec0-7f95-4853-a9ad-061989f26c8a

**Comment by xwjiang2010** [comment]
test is hanging. Could that have something to do with the fact that Trainer also reserves one CPU? So 8 CPU won't cut it?

**Comment by xwjiang2010** [comment]
https://buildkite.com/ray-project/release-tests-pr/builds/8988#0181f2e0-c269-4390-9f91-6d5712eb06d9

File: release/train_tests/horovod/train_horovod_multi_node_test.py [hunk 1]
```diff
@@ -3,28 +3,30 @@
...
+    num_workers = 7  # 1 reserved for trainable
     num_epochs = 10
-    trainer = Trainer("horovod", num_workers)
-    trainer.start()
-    results = trainer.run(
-        horovod_torch_train_func, config={"num_epochs": num_epochs, "lr": 1e-3}
+    trainer = HorovodTrainer(
+        horovod_torch_train_func,
+        train_loop_config={"num_epochs": num_epochs, "lr": 1e-3},
+        scaling_config=dict(num_workers=num_workers),
```
[the remark this answers is not in the document]
Comment (Yard1) [reply release/train_tests/horovod/train_horovod_multi_node_test.py:21]: sure - what difference would that make though?

[the remark this answers is not in the document]
Comment (krfricke) [reply release/train_tests/horovod/train_horovod_multi_node_test.py:21]: We can still use `num_workers = 8`

**Review by krfricke — APPROVED** [review]

**Pull request merged**